src/main/webapp/mapScript.js

```diff
@@ -5,39 +5,88 @@ goog.require('templates.infoWindow');
-let groupArray = []; 
+let groupMarkers = new Map(); 
+
+// Default map placement centered at Mountain View, CA
+const DEFAULT_LAT = 37.3868;
+const DEFAULT_LNG = -122.085;
+const DEFAULT_ZOOM = 10;
+ 
+let map; 
 
 /**
  * Initialize the map to show Mountain View and events in that area
  */
 function initMap() {
-  
-    // Coordinates for the default map center of Mountain View, CA
-    const mapViewDefault = {lat: 37.3868, lng: -122.085}; 
+    const mapViewDefault = {lat: DEFAULT_LAT, lng: DEFAULT_LNG}; 
 
-    const map = new google.maps.Map(goog.dom.getElement('map'), {
-    // Zoom set to 8 as default until the radius is set up with events
-        zoom: 8,
+    map = new google.maps.Map(goog.dom.getElement('map'), {
+        // Zoom level will change automatically with user's search
+        zoom: DEFAULT_ZOOM,
         center: mapViewDefault,
     });
 
     const geocoder = new google.maps.Geocoder();
-    // TODO: repace the hardcoded event ID with 
-    getMarkerInfo(123, geocoder, map);
+    // TODO: repace the hardcoded event ID with actually value
+    getMarkerInfo('123', geocoder);
+
+
+    autoCompleteAndZoom(); 
+
     document.getElementById('submit').addEventListener('click', () => {
-        geocodeAddress(geocoder, map);
+        geocodeAddress(geocoder);
     });
+
 }
 
+/**
+ * Adds the auto Complete for the address search bar 
+ * Allows user to hit enter instead of button
+ * Automatically zooms according to the location inputed
+ */
+function autoCompleteAndZoom() {
+    const address = document.getElementById('address');
+    const searchBox = new google.maps.places.SearchBox(address);
+
+    map.addListener("bounds_changed", function() {
+        searchBox.setBounds(map.getBounds());
+    });
+
+    searchBox.addListener("places_changed", function() {
+        const places = searchBox.getPlaces();
+
+        // Checks if there are not autocomplete options
+        if (places.length == 0) {
+            return;
+        }
+
+        let bounds = new google.maps.LatLngBounds();
+        places.forEach(function(place) {
+            if (!place.geometry) {
+                console.log("Returned place contains no geometry");
+                return;
+            }
+
+            // Get the bounds for the location
+            if (place.geometry.viewport) {
+                bounds.union(place.geometry.viewport);
+            } else {
+                bounds.extend(place.geometry.location);
+            }
+        });
+        map.fitBounds(bounds);
+    }); 
+}
 
 /**
  * Finds the location on the map for the given address
  */
-function geocodeAddress(geocoder, resultsMap) {
+function geocodeAddress(geocoder) {
     const address = document.getElementById('address').value;
+
     geocoder.geocode({'address': address}, (results, status) => {
         if (status === 'OK') {
-            resultsMap.setCenter(results[0].geometry.location);
+            map.setCenter(results[0].geometry.location);
```

Comment:
Do you need to error check for if results is empty? else results[0] may crash

Reply:
If the results are empty the status would not be OK and there would be an alert that says: "Geocode was not successful for the following reason: ZERO_RESULTS"